src/fmu/dataio/providers/_objectdata_base.py

```diff
@@ -0,0 +1,393 @@
+            variable is set for those who wants it turned around).
+        """
+
+        tdata = self.dataio.timedata
+        use_legacy_format: bool = self.dataio.legacy_time_format
+
+        if not tdata:
+            return None
+
+        if len(tdata) == 1:
+            start = TimedataValueLabel.from_list(tdata[0])
+            self.time0 = start.value
+            return (
+                TimedataLegacyFormat([start])
+                if use_legacy_format
+                else TimedataFormat(start, None)
+            )
+
+        if len(tdata) == 2:
+            start, stop = (
+                TimedataValueLabel.from_list(tdata[0]),
+                TimedataValueLabel.from_list(tdata[1]),
+            )
+
+            if datetime.fromisoformat(start.value) > datetime.fromisoformat(stop.value):
+                start, stop = stop, start
+
+            self.time0, self.time1 = start.value, stop.value
+
+            return (
+                TimedataLegacyFormat([start, stop])
+                if use_legacy_format
+                else TimedataFormat(start, stop)
+            )
+
+        return (
+            TimedataLegacyFormat([])
+            if use_legacy_format
+            else TimedataFormat(None, None)
+        )
+
+    def _derive_from_existing(self) -> None:
+        """Derive from existing metadata."""
+
+        # do not change any items in 'data' block, as it may ruin e.g. stratigrapical
+        # setting (i.e. changing data.name is not allowed)
+        self.metadata = self.meta_existing["data"]
+        self.name = self.meta_existing["data"]["name"]
+
+        # derive the additional attributes needed later e.g. in Filedata provider:
+        relpath = Path(self.meta_existing["file"]["relative_path"])
+        if self.dataio.subfolder:
+            self.efolder = relpath.parent.parent.name
+        else:
+            self.efolder = relpath.parent.name
+
+        self.classname = self.meta_existing["class"]
+        self.extension = relpath.suffix
+        self.fmt = self.meta_existing["data"]["format"]
+
+        # TODO: Clean up types below.
+        self.time0, self.time1 = parse_timedata(self.meta_existing["data"])  # type: ignore
+
+    def derive_metadata(self) -> None:
+        """Main function here, will populate the metadata block for 'data'."""
+        logger.info("Derive all metadata for data object...")
+
+        if self.meta_existing:
+            self._derive_from_existing()
+            return
+
+        namedstratigraphy = self._derive_name_stratigraphy()
+        objres = self._derive_objectdata()
+        if self.dataio.forcefolder and not self.dataio.forcefolder.startswith("/"):
+            msg = (
+                f"The standard folder name is overrided from {objres.efolder} to "
+                f"{self.dataio.forcefolder}"
+            )
+            objres.efolder = self.dataio.forcefolder
+            logger.info(msg)
+            warn(msg, UserWarning)
```

Comment:
Since it's a bit hard to understand what might have changed in this class from the diff ....

This used to be at the bottom of `_derive_objectdata`